File: problems/sorting/LC169_majority_elements/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
  int majorityElement(std::vector<int>& nums) {
    sort(nums.begin(), nums.end());
    int n = nums.size();
    return nums[n/2];
  }
};

int main() {
  Solution s = Solution();
  std::vector<int> nums = {2,2,1,1,1,2,2};
    std::cout << s.majorityElement(nums) << std::endl;
}

File: problems/sorting/LC169_majority_elements/Solution.java
package problems.sorting.LC169_majority_elements;
import java.util.Arrays;

public class Solution {
  int majorityElement(int[] nums) {
    Arrays.sort(nums);
    int n = nums.length;
    return nums[n/2];
  }

  public static void main(String[] args) {
    int[] nums = new int[]{2,2,1,1,1,2,2};
    Solution s = new Solution();
    int majorityNum = s.majorityElement(nums);
    System.out.println("majority: " + majorityNum);
  }
}
